### DISCUSSION TITLE
API limits

### DISCUSSION THREAD

**Discussion by danielcardeenas** [cloudbet/docs#36] [General]
What are the limits for Cloudbet public api?
I mean, rps, rpm, etc...

**Comment by kgravenreuth** [comment]
Hi @danielcardeenas 

The [Cloudbet Public Feed API](https://www.cloudbet.com/api/?urls.primaryName=Feed) currently does not have strict rate limits (although this may change in the future). We request you to hit our API with reasonable frequency without abusing our system.

On the other hand, the [Cloudbet Trading API](https://www.cloudbet.com/api/?urls.primaryName=Trading) does have a rate limit as mentioned in the linked API docs: 

```
Rate limit
Our place bet endpoint has rate limit - 1 request per second (1 RPS), we provide headers in response X-RateLimit-Limit and X-RateLimit-Remaining. Exceeding this rate limit will return HTTP 429 Too Many Requests response status code error.
```

Furthermore, please place bets on the Trading API with reasonable odds as per our Feed API responses. For example, if 10s to hundreds of bets are continuously placed which repeatedly get rejected due to invalid odds in the request, then we reserve the right to take action on accounts for stability of our systems. This should usually not happen if bets are placed based on up to date Feed API responses.

Please let us know if you have further questions.

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by danielcardeenas** [reply]
Thanks a lot
